packages/rln/src/contract/constants.ts

```diff
@@ -1,68 +1,35 @@
-  "function transferOwnership(address newOwner)",
-  "function upgradeTo(address newImplementation)",
-  "function upgradeToAndCall(address newImplementation, bytes data) payable",
-  "function usingStorageIndex() view returns (uint16)"
-];
-
-// ref https://github.com/waku-org/waku-rln-contract/blob/19fded82bca07e7b535b429dc507cfb83f10dfcf/deployments/sepolia/WakuRlnStorage_0.json#L3
-export const RLN_STORAGE_ABI = [
-  "constructor(address _poseidonHasher, uint16 _contractIndex)",
-  "error DuplicateIdCommitment()",
-  "error FullTree()",
-  "error InvalidIdCommitment(uint256 idCommitment)",
-  "error NotImplemented()",
-  "event MemberRegistered(uint256 idCommitment, uint256 index)",
-  "event MemberWithdrawn(uint256 idCommitment, uint256 index)",
-  "event OwnershipTransferred(address indexed previousOwner, address indexed newOwner)",
-  "function DEPTH() view returns (uint256)",
-  "function MEMBERSHIP_DEPOSIT() view returns (uint256)",
-  "function SET_SIZE() view returns (uint256)",
-  "function contractIndex() view returns (uint16)",
-  "function deployedBlockNumber() view returns (uint32)",
-  "function idCommitmentIndex() view returns (uint256)",
-  "function isValidCommitment(uint256 idCommitment) view returns (bool)",
-  "function memberExists(uint256) view returns (bool)",
-  "function members(uint256) view returns (uint256)",
-  "function owner() view returns (address)",
-  "function poseidonHasher() view returns (address)",
-  "function register(uint256[] idCommitments)",
-  "function register(uint256 idCommitment) payable",
-  "function renounceOwnership()",
-  "function slash(uint256 idCommitment, address receiver, uint256[8] proof) pure",
-  "function stakedAmounts(uint256) view returns (uint256)",
-  "function transferOwnership(address newOwner)",
-  "function verifier() view returns (address)",
-  "function withdraw() pure",
-  "function withdrawalBalance(address) view returns (uint256)"
-];
+import { RLN_V2_ABI } from "./abi/rlnv2.js";
 
 export const SEPOLIA_CONTRACT = {
   chainId: 11155111,
-  address: "0xF471d71E9b1455bBF4b85d475afb9BB0954A29c4",
-  abi: RLN_REGISTRY_ABI
+  address: "0xCB33Aa5B38d79E3D9Fa8B10afF38AA201399a7e3",
+  abi: RLN_V2_ABI
 };
+
+/**
+ * Rate limit tiers (messages per epoch)
+ * Each membership can specify a rate limit within these bounds.
+ * @see https://github.com/waku-org/specs/blob/master/standards/core/rln-contract.md#implementation-suggestions
+ */
+export const RATE_LIMIT_TIERS = {
+  LOW: 20, // Suggested minimum rate - 20 messages per epoch
+  MEDIUM: 200,
+  HIGH: 600 // Suggested maximum rate - 600 messages per epoch
+} as const;
+
+// Default to maximum rate limit if not specified
+export const DEFAULT_RATE_LIMIT = RATE_LIMIT_TIERS.HIGH;
+
+/**
+ * Epoch length in seconds (10 minutes)
+ * This is a constant defined in the smart contract
+ */
+export const EPOCH_LENGTH = 600;
+
+// Global rate limit parameters
+export const RATE_LIMIT_PARAMS = {
```

Comment:
As I see from the code `DEFAULT_RATE_LIMIT` and `EPOCH_LENGTH` can be easily removed as independent variables and can be used directly from `RATE_LIMIT_PARAMS` which essentially contains default values.

Reply:
updated